server/src/main/java/org/opensearch/node/remotestore/RemoteStoreNodeAttribute.java

```diff
@@ -56,8 +63,18 @@ public class RemoteStoreNodeAttribute {
     /**
      * Creates a new {@link RemoteStoreNodeAttribute}
      */
-    public RemoteStoreNodeAttribute(DiscoveryNode node) {
+    public RemoteStoreNodeAttribute(DiscoveryNode node, Settings settings) {
```

Comment:
could we have just passed a boolean output of `isRemoteRoutingTableEnabled` whenever this class is initialized instead of the entire settings object? as i see we are only using this in ctor

Reply:
This is to ensure any further feature flags requiring new repos can add similar handling.

Reply:
This is now removed, handling this through JoinTaskExecutor